File: application/query/request.go
package query

type Request struct {
	ID       int64     `json:"id"`
	Current  int       `json:"current"`
	PageSize int       `json:"pageSize" binding:"max=1000"`
	OrderBy  []OrderBy `json:"orderBy"`
	QueryBy  []QueryBy `json:"queryBy"`
	Omit     string    `json:"omit"`
	Keyword  string    `json:"keyword"`
}

func NewRequest(args ...int) Request {
	req := Request{
		Current:  1,
		PageSize: 10,
	}
	if len(args) > 0 {
		req.Current = args[0]
	}
	if len(args) > 1 {
		req.PageSize = args[1]
	}
	return req
}

type OrderBy struct {
	Field string `json:"field"`
	Value string `json:"value"` // asc desc
}

type QueryBy struct {
	Field string `json:"field"`
	Value any    `json:"value"`
}

File: web/packages/adm/src/service/model/request.ts
export type OrderBy = {
    field: string,
    asc: boolean
}

export type QueryBy = {
    field: string,
    value: any
}

export class Request {
    current?: number;
    pageSize?: number;
    orderBy?: OrderBy[];
    queryBy?: QueryBy[];

    constructor(pagination?: { current: number, pageSize: number }) {
        if (!!pagination) {
            this.current = pagination.current;
            this.pageSize = pagination.pageSize;
        }
    }
}
